### PR TITLE
Drop the auth-DB remove in test-esx.

### PR DESCRIPTION
**Pull request by lipingxue**
This PR includes two changes:
1. drop the auth-DB remove in test-esx (fixes issue #875)
2. Fix a small bug in function "get_datastore_url" and "get_datastore_name" to handle the _DEFAULT_DS correctly.

test-esx passed. Please review this change.

### CODE REVIEW

**Comment by shaominchen** [comment]
LGTM.

File: esx_service/utils/auth_data.py [hunk 1]
```diff
@@ -210,6 +210,7 @@ def get_default_datastore(self, conn):
             return str(error_info), None
         else:
             datastore_url = result[auth_data_const.COL_DEFAULT_DATASTORE_URL]
+            logging.debug("get_default_datastore: datastore_url=%s", datastore_url)
```
Comment (msterin) [inline esx_service/utils/auth_data.py]: nit - why this is dropped ? Looks like a useful debug info for me. 
It's a nit, feel free to merge with it

File: misc/drone-scripts/validate-and-serialize.sh [hunk 2]
```diff
@@ -12,15 +12,15 @@
...
+DS='-ds=datastore1'
 unit_test_array=($TEST_URL_ARRAY)
 numServers=${#unit_test_array[@]}
 DRONE_BUILD_NUMBER=${DRONE_BUILD_NUMBER:=0}
 prevBuildStatus=`drone build info vmware/docker-volume-vsphere $(( $DRONE_BUILD_NUMBER-$numServers ))`
 outArray=($prevBuildStatus)
 total_builds=`drone build list vmware/docker-volume-vsphere| wc -l`
 
-govc datastore.mkdir docker-volume-vsphere/$DRONE_BUILD_NUMBER
+govc datastore.mkdir $DS docker-volume-vsphere/$DRONE_BUILD_NUMBER
```
Comment (msterin) [inline misc/drone-scripts/validate-and-serialize.sh]: so on which DS it was doing mkdir before the change ?

File: esx_service/utils/auth_data.py (hunk 1 above)
Comment (lipingxue) [inline esx_service/utils/auth_data.py]: It is NOT dropped. I added this logging in my change.

Comment (lipingxue) [inline esx_service/utils/auth_data.py]: It is NOT dropped, I added this log in my change :)

File: misc/drone-scripts/validate-and-serialize.sh (hunk 2 above)
Comment (lipingxue) [inline misc/drone-scripts/validate-and-serialize.sh]: This change is not from me. It was made by Ritesh in master, I rebase my change with master and then you see the change. I have removed from my commit log.

File: esx_service/utils/auth_data.py (hunk 1 above)
Comment (msterin) [inline esx_service/utils/auth_data.py]: oops. you are right :-) !

**Pull request merged**